.github/workflows/java-integration-tests.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checks that are intended to run on PRs containing Java code.
+
+name: Java Integration Tests
```

Comment:
Taking a step back, what is your goal with these workflows? The Java PR workflow already can be run manually, why is having separate workflows better?

It seems like maybe the goal is to have a more granular signal? Or to allow individual pieces to be run independently of the main workflow?

I think there might be better ways to accomplish this - having 2 workflow files that do more or less the same thing is somewhat of an antipattern (and is just expensive). Rather than doing this, maybe we can restructure the existing java pr workflow to have some conditional logic. For example:

1) Always run everything on schedule
2) Run everything after build in parallel instead of serially
3) Insert if conditions on the job that are basically like: `if: not is(workflow_dispatch) || input.run_its` (similar to https://github.com/apache/beam/blob/ea3c2a4dbbe20a8e7161d362102bb1753bd60bba/.github/workflows/build_release_candidate.yml#L25)

Reply:
I'm generally supportive of moving things to composite actions where helpful, but I don't like doubling our workflow count, especially since we'd now need to do more work to keep different workflows in sync.